### ISSUE TITLE
Provide a Resource API Helper and example for testing Application Insights

### ISSUE THREAD

**Issue by paul-warwick-cdt** [invictus-integration/testing-framework#60]
**This PR feature will be submitted to supply a generic Helper provider for tests that access Azure query or run instructions through an API.  The example will be Application Insights

**Describe the solution you'd like**
This solution will be an enhancement to the Invictus Testing Framework
This solution will provide base access to tests through https://management.azure.com
This solution will provide base access to tests through https://api.loganalytics.io
This solution will provide base access to tests through https://api.applicationinsights.io

**Describe alternatives you've considered**
This solution provides the reliable API experience for testing with c# providers where c# base classes are missing or complex.

**Additional context**
This feature will provide example narrative for tests on Application Insights on https://api.applicationinsights.io
the contexts to be covered will be:
Authentication, Controlling a single Run, Retrieving runs by Polling

**Comment by paul-warwick-cdt** [comment]
[AppInsightsInvictusWebGuide.docx](https://github.com/invictus-integration/testing-framework/files/5196230/AppInsightsInvictusWebGuide.docx)
Attaching proposal narrative

**Comment by tomkerkhove** [comment]
I think we should hold off the PR until we come to an agreement on the specs here because the current approach is not really inline with our current approach.

Things that I would encourage to change:
- Remove the new authentication and refactor to the current one which we'll rename to `AzureAuthentication`(?)
- `ResourceApiProvider` is not really clear and instead we try to provide providers per service, for example `ApplicationInsightsProvider`

So putting it together, it would look similar to this:
```csharp
var authentication = AzureAuthentication.UsingServicePrincipal(tenantId, subscriptionId, clientId, clientSecret);

// Create provider
var provider = ApplicationInsightsProvider.LocatedAt(resourceGroupName, applicationInsightsName, authentication);

// For the client
using (var applicationInsights = await ApplicationInsightsClient.CreateAsync(resourceGroup, applicationInsightsName, authentication))    
{    
     applicationInsights.Something()
}
```

However, the things I see in the samples above are more related to Log Analtyics maybe? Or what is the goal there? It's not really clear what the use cases would be.

**Comment by paul-warwick-cdt** [comment]
Agreed this is very much a base class (bottom of 3 tiers) that sits below the "Step Definitions" mid-layer, and which in a SpecFlow solution gives the kinds of scenario/encapsulation for the suggested samples.

**Comment by tomkerkhove** [comment]
I'd suggest that we do this gradually then and implement an additional provider without the additional abstraction with a follow-up PR to refactor LA + new provider into that model, or combine it in the same one but not just a PR with the foundation as it's not easy to grasp if the approach is correct for us

**Comment by paul-warwick-cdt** [comment]
[Invictus_Refactor_proposal.pptx](https://github.com/invictus-integration/testing-framework/files/5400257/Invictus_Refactor_proposal.pptx)
[AppInsightsInvictusGuide_60_2.docx](https://github.com/invictus-integration/testing-framework/files/5400258/AppInsightsInvictusGuide_60_2.docx)

Following review attached are:
[1] Powerpoint of a 1 hour refactor proposed to separate common code for Authentication and allow this to be leveraged both by Logic Apps & a new Invictus Authentication class re-usable across all API related future features
[2] A simplified word doc version of the starter Application Insights provider overview for future review/implementation
